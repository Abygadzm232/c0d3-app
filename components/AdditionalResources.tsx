import React from 'react'
import '../scss/additionalResources.scss'

const AdditionalResources: React.FC = () => {
  return (
    <div className="additional-resources__container mt-4 mb-0">
      <h1 className="additional-resources__title text-muted">
        ADDITIONAL RESOURCES
      </h1>
      <hr className="my-2" />
      <a
        rel="noopener noreferrer"
        target="_blank"
        className="mx-3 "
        href="https://www.notion.so/Setup-Instructions-fc8f8fcc1376482ead839fa6b1034cb4"
      >
        Setup Instructions
      </a>
      <hr className="my-2" />
      <a className="mx-3 " href="/">
        Github Basics
      </a>
      <hr className="my-2" />
      <a
        className="mx-3"
        href="https://stackoverflow.com/"
        target="_blank"
        rel="noopener noreferrer"
      >
        Stack Overflow
      </a>
      <hr className="my-2" />
      <a
        className="mx-3"
        href="https://chat.c0d3.com/"
        target="_blank"
        rel="noopener noreferrer"
      >
        Help & Support
      </a>
      <hr className="my-2" />
    </div>
  )
}

export default AdditionalResources
