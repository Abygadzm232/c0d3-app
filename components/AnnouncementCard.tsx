import React from 'react'

import '../scss/announcementCard.scss'

type Props = {
  announcementOne: string
  announcementTwo: string
  announcementThree: string
  announcementFour: string
}

const AnnouncementCard: React.FC<Props> = ({
  announcementOne,
  announcementTwo,
  announcementThree,
  announcementFour
}) => {
  return (
    <div className="announcement-card__container card shadow-sm mt-3 p-2 d-flex border-0">
      <div className="card-body">
        <h1 className="announcement-card__title">General Announcements</h1>
        <div className="announcement-card__paragraph-container mt-3">
          <h2 className="announcement-card__subtitle font-weight-bold">
            Updates and Guidelines
          </h2>
          <p className="mt-4">{announcementOne}</p>
          <p className="mt-4">{announcementTwo}</p>
          <p className="mt-4">{announcementThree}</p>
          <p className="mt-4">{announcementFour}</p>
        </div>
      </div>
    </div>
  )
}

export default AnnouncementCard
