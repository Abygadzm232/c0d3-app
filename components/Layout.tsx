import React from 'react'
import AppNav from './AppNav'
import Footer from './Footer'
import SessionContext from '../helpers/contexts/session'
import _ from 'lodash'

type Props = {
  children: React.ReactElement
}

const Layout: React.FC<Props> = ({ children }) => {
  const { session } = React.useContext(SessionContext)

  if (session && session.user) {
    return (
      <>
        <AppNav username={session.user.username} loggedIn />
        <div className="container">{children}</div>
        <Footer />
      </>
    )
  }

  return (
    <>
      <AppNav />
      <div className="container">{children}</div>
      <Footer />
    </>
  )
}

export default Layout
