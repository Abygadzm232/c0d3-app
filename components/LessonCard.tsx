import React from 'react'
import { CheckCircle } from 'react-feather'
import '../scss/lessonCard.scss'

type Props = {
  coverImg: string
  title: string
  challengeCount: number
  description: string
  currentState?: string
  reviewUrl: string
}

const LessonCard: React.FC<Props> = props => {
  const containerClass =
    props.currentState === 'inProgress'
      ? 'lesson-card__container_inprogress border-primary'
      : ''
  return (
    <div className={`card shadow-sm mt-3 border-0 ${containerClass}`}>
      <div className="card-body p-2">
        <div className="row no-gutters">
          {props.currentState === 'completed' && (
            <span className="badge badge-pill badge-success position-absolute lesson-card__container_completed">
              <CheckCircle style={{ height: '15px', marginTop: '-2px' }} />
              <span className="mr-1">COMPLETED</span>
            </span>
          )}
          <div className="mw-100 col-2 mr-4">
            <img
              src={`/curriculumAssets/lessonCoversSvg/${props.coverImg}`}
              alt={props.coverImg}
            />
          </div>
          <div className="col-9">
            <h4 className="lesson-card__title font-weight-bold mt-3">
              {props.title}
            </h4>
            <div>
              <div className="d-inline-block mr-4">
                <img
                  className="mr-1"
                  src="/curriculumAssets/icons/icon-challenge.svg"
                  alt="icon-challenge"
                />
                <span className="lesson-card__icon-text">
                  {props.challengeCount} CHALLENGES
                </span>
              </div>
            </div>
            <p className="lesson-card__description mt-2">{props.description}</p>
          </div>
          {props.currentState === 'completed' && (
            <span className="position-absolute lesson-card__container_review">
              <a
                href={props.reviewUrl}
                className="btn btn-sm bg-primary text-white"
              >
                Review Submissions
              </a>
            </span>
          )}
        </div>
      </div>
      {props.currentState === 'inProgress' && (
        <div className="card-footer bg-primary">
          <a
            className="lesson-card__button btn btn-light mr-2 my-1 text-primary"
            href="#"
          >
            Start Lesson
          </a>
          <a
            className="lesson-card__button btn bg-primary my-1 text-white border border-white"
            href="#"
          >
            View Challenges
          </a>
        </div>
      )}
    </div>
  )
}

export default LessonCard
