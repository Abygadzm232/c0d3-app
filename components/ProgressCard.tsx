import React from 'react'
import NavLink from './NavLink'

import '../scss/progressCard.scss'

type Props = {
  progressCount: number
}

const ProgressSVG: React.FC<Props> = ({ progressCount }) => {
  let x = 15
  if (progressCount === 100) x = 9
  else if (progressCount >= 10) x = 11

  return (
    <svg viewBox="0 0 40 40" width="65px">
      <circle
        cx="20"
        cy="20"
        r="17"
        fill="transparent"
        stroke="#3a22b8"
        strokeWidth="4.5"
      />
      <circle
        cx="20"
        cy="20"
        r="17"
        fill="transparent"
        stroke="#fff"
        strokeDasharray={`${progressCount} ${100 - progressCount}`}
        strokeDashoffset="20"
        strokeWidth="4.5"
      />
      <g className="circle-label">
        <text
          color="#fff"
          x={x}
          y="23"
          fill="#fff"
          style={{ font: 'bold 10px sans-serif' }}
        >
          {progressCount}
          <tspan fontSize="8px">%</tspan>
        </text>
      </g>
    </svg>
  )
}

const ProgressCard: React.FC<Props> = ({ progressCount }) => {
  return (
    <div className="progress-card__container d-flex card shadow-sm mt-3 bg-primary text-white p-2 border-0">
      <div className="card-body">
        <ProgressSVG progressCount={progressCount} />
        <h4 className="progress-card__title mt-3">
          You&#39;re off to a great start!
        </h4>
        <div className="mt-3">
          <p>
            Start by setting up your development environment and then follow the
            lessons in the course.
          </p>
        </div>
        <NavLink
          path="https://www.notion.so/Setup-Instructions-fc8f8fcc1376482ead839fa6b1034cb4#10ab6286f4d942c58bd330f68cb8cb95"
          className="btn btn-light progress-card__button mt-2 text-primary"
          external
        >
          Setup Development Environment
        </NavLink>
      </div>
    </div>
  )
}

export default ProgressCard
