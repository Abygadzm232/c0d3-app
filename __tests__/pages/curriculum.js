jest.mock('@apollo/react-hooks')
jest.mock('../../components/LessonCard')
import React from 'react'
import { useQuery } from '@apollo/react-hooks'
import Curriculum from '../../pages/curriculum'
import { render } from '@testing-library/react'
import LessonCard from '../../components/LessonCard'
import Layout from '../../components/Layout'
import ProgressCard from '../../components/ProgressCard'
import AnnouncementCard from '../../components/AnnouncementCard'
import AdditionalResources from '../../components/AdditionalResources'

describe('Curriculum Page', () => {
  test('Should render data with lesson state completed', async () => {
    useQuery.mockReturnValue({
      data: {
        lessons: [
          {
            challenges: [{ id: '4' }],
            id: '4'
          }
        ],
        session: {
          user: {
            username: 'test'
          },
          lessonStatus: [
            {
              lessonId: '4',
              isPassed: '1235435',
              isEnrolled: '123456'
            }
          ]
        }
      }
    })
    LessonCard.mockReturnValue(<h1>LessonCard</h1>)
    const { container, debug } = render(<Curriculum />)
    expect(container).toMatchSnapshot()
  })

  test('Should render data with lesson state inProgress', async () => {
    useQuery.mockReturnValue({
      data: {
        lessons: [
          {
            challenges: [{ id: '4' }],
            id: '4'
          }
        ],
        session: {
          user: {
            username: 'test'
          },
          lessonStatus: [
            {
              lessonId: '4',
              isEnrolled: '123456'
            }
          ]
        }
      }
    })
    LessonCard.mockReturnValue(<h1>LessonCard</h1>)
    const { container, debug } = render(<Curriculum />)
    expect(container).toMatchSnapshot()
  })

  test('Should render render first lesson when lessonStatus is empty', async () => {
    useQuery.mockReturnValue({
      data: {
        lessons: [
          {
            challenges: [{ id: '4' }],
            id: '4'
          }
        ],
        session: {
          user: {
            username: 'test'
          },
          lessonStatus: []
        }
      }
    })
    LessonCard.mockReturnValue(<h1>LessonCard</h1>)
    const { container, debug } = render(<Curriculum />)
    expect(container).toMatchSnapshot()
  })
  /*
  test('Should render data with lesson state inProgress', async () => {
    useQuery.mockReturnValue({
      data: {
        curriculumStatus: [
          {
            id: 4,
            title: 'lesson 4',
            description: '4th lesson',
            order: 4,
            challenges: [
              {
                id: 4
              }
            ],
            currentUser: {
              userLesson: {
                isPassed: '234234'
              }
            },
            docUrl: 'lesson4.com'
          },
          {
            id: 1,
            title: 'lesson 1',
            description: '1st lesson',
            order: 1,
            challenges: [
              {
                id: 1
              }
            ],
            currentUser: {
              userLesson: {
                isPassed: '9878990'
              }
            },
            docUrl: 'lesson1.com'
          }
        ]
      }
    })
    LessonCard.mockReturnValue(<h1>LessonCard</h1>)

    const { container } = render(<Curriculum />)
    expect(container).toMatchSnapshot()
  })*/
})
