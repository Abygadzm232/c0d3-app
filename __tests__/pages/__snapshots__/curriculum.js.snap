// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Curriculum Page Should render data with lesson state completed 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light justify-content-between bg-white"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand text-primary font-weight-bold"
        href="/#"
      >
        C0D3
      </a>
      <div
        id="navbarNav"
      >
        <div
          class="navbar-nav collapse navbar-collapse"
        >
          <div
            class="navbar-nav collapse navbar-collapse"
          >
            <a
              class="nav-item nav-link active"
              href="/#"
            >
              Home 
              <span
                class="sr-only"
              >
                (current)
              </span>
            </a>
            <a
              class="nav-item nav-link"
              href="/#learning"
            >
              Learning Process
            </a>
            <a
              class="nav-item nav-link"
              href="https://c0d3.com/book"
            >
              Resources
            </a>
            <a
              class="nav-item nav-link"
              href="https://chat.c0d3.com"
            >
              Help
            </a>
          </div>
        </div>
      </div>
      <div>
        <a
          class="btn btn-secondary border mr-3"
          href="/login"
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  <div
    class="container"
  >
    <div
      class="row mt-4"
    >
      <div
        class="col-8"
      >
        <div
          class="card shadow-sm mt-3 border-0 lesson-card__container_inprogress border-primary"
        >
          <div
            class="card-body p-2"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="mw-100 col-2 mr-4"
              >
                <img
                  alt="js-0-cover.svg"
                  src="/curriculumAssets/lessonCoversSvg/js-0-cover.svg"
                />
              </div>
              <div
                class="col-9"
              >
                <h4
                  class="lesson-card__title font-weight-bold mt-3"
                >
                  lesson 1
                </h4>
                <div>
                  <div
                    class="d-inline-block mr-4"
                  >
                    <img
                      alt="icon-challenge"
                      class="mr-1"
                      src="/curriculumAssets/icons/icon-challenge.svg"
                    />
                    <span
                      class="lesson-card__icon-text"
                    >
                      1
                       CHALLENGES
                    </span>
                  </div>
                </div>
                <p
                  class="lesson-card__description mt-2"
                >
                  1st lesson
                </p>
              </div>
            </div>
          </div>
          <div
            class="card-footer bg-primary"
          >
            <a
              class="lesson-card__button btn btn-light mr-2 my-1 text-primary"
              href="lesson1.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              Start Lesson
            </a>
            <a
              class="lesson-card__button btn bg-primary my-1 text-white border border-white"
              href="#"
            >
              View Challenges
            </a>
          </div>
        </div>
        <div
          class="card shadow-sm mt-3 border-0 "
        >
          <div
            class="card-body p-2"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="mw-100 col-2 mr-4"
              >
                <img
                  alt="js-1-cover.svg"
                  src="/curriculumAssets/lessonCoversSvg/js-1-cover.svg"
                />
              </div>
              <div
                class="col-9"
              >
                <h4
                  class="lesson-card__title font-weight-bold mt-3"
                >
                  lesson 4
                </h4>
                <div>
                  <div
                    class="d-inline-block mr-4"
                  >
                    <img
                      alt="icon-challenge"
                      class="mr-1"
                      src="/curriculumAssets/icons/icon-challenge.svg"
                    />
                    <span
                      class="lesson-card__icon-text"
                    >
                      1
                       CHALLENGES
                    </span>
                  </div>
                </div>
                <p
                  class="lesson-card__description mt-2"
                >
                  4th lesson
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-4"
      >
        <div
          class="progress-card__container d-flex card shadow-sm mt-3 bg-primary text-white p-2 border-0"
        >
          <div
            class="card-body"
          >
            <div
              class="progress-card__progress-bar d-flex justify-content-center align-items-center rounded-circle"
            >
              <span
                class="font-weight-bold"
              >
                0
                %
              </span>
            </div>
            <h4
              class="progress-card__title mt-3"
            >
              You're off to a great start!
            </h4>
            <div
              class="mt-3"
            >
              <p>
                Start by setting up your development environment and then follow the lessons in the course.
              </p>
            </div>
            <a
              class="btn btn-light progress-card__button mt-2 text-primary"
              href="https://www.notion.so/Setup-Instructions-fc8f8fcc1376482ead839fa6b1034cb4#10ab6286f4d942c58bd330f68cb8cb95"
            >
              Setup Development Environment
            </a>
          </div>
        </div>
        <div
          class="announcement-card__container card shadow-sm mt-3 p-2 d-flex border-0"
        >
          <div
            class="card-body"
          >
            <h1
              class="announcement-card__title"
            >
              General Announcements
            </h1>
            <div
              class="mt-3"
            >
              <p
                class="font-weight-bold"
              >
                Updates and Guidelines
              </p>
              <p
                class="mt-4"
              >
                To make space for other students on our servers, your account will be deleted after 30 days of inactivity.
              </p>
              <p
                class="mt-4"
              >
                Take each lesson challenge seriously and do them over and over again until you can solve them. With the exception End to End, all challenges are questions and exercises taken from real interviews.
              </p>
              <p
                class="mt-4"
              >
                This lesson will not only prepare you for interviews, but it will also help teach you the skills that you need to become an effective engineer.
              </p>
              <p
                class="mt-4"
              >
                After completing Foundations of JavaScript, Variables & Functions, Array, Objects, End to End, HTML/CSS/JavaScript, React/GraphQL/SocketIO, you will be technically ready to contribute to our codebase.
              </p>
            </div>
          </div>
        </div>
        <div
          class="additional-resources__container mt-4 mb-0"
        >
          <h1
            class="additional-resources__title text-muted"
          >
            ADDITIONAL RESOURCES
          </h1>
          <hr
            class="my-2"
          />
          <a
            class="mx-3 "
            href="https://www.notion.so/Setup-Instructions-fc8f8fcc1376482ead839fa6b1034cb4"
          >
            Setup Instructions
          </a>
          <hr
            class="my-2"
          />
          <a
            class="mx-3 "
            href="#"
          >
            Github Basics
          </a>
          <hr
            class="my-2"
          />
          <a
            class="mx-3 "
            href="https://stackoverflow.com/"
          >
            Stack Overflow
          </a>
          <hr
            class="my-2"
          />
          <a
            class="mx-3 "
            href="https://chat.c0d3.com/"
          >
            Help & Support
          </a>
          <hr
            class="my-2"
          />
        </div>
      </div>
    </div>
  </div>
  <footer
    class="mt-5 font-weight-light text-center pb-5 text-muted"
  >
    Â© Copyright 2020 GarageScript
    <br />
    All Rights Reserved
  </footer>
</div>
`;

exports[`Curriculum Page Should render data with lesson state inProgress 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light justify-content-between bg-white"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand text-primary font-weight-bold"
        href="/#"
      >
        C0D3
      </a>
      <div
        id="navbarNav"
      >
        <div
          class="navbar-nav collapse navbar-collapse"
        >
          <div
            class="navbar-nav collapse navbar-collapse"
          >
            <a
              class="nav-item nav-link active"
              href="/#"
            >
              Home 
              <span
                class="sr-only"
              >
                (current)
              </span>
            </a>
            <a
              class="nav-item nav-link"
              href="/#learning"
            >
              Learning Process
            </a>
            <a
              class="nav-item nav-link"
              href="https://c0d3.com/book"
            >
              Resources
            </a>
            <a
              class="nav-item nav-link"
              href="https://chat.c0d3.com"
            >
              Help
            </a>
          </div>
        </div>
      </div>
      <div>
        <a
          class="btn btn-secondary border mr-3"
          href="/login"
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  <div
    class="container"
  >
    <div
      class="row mt-4"
    >
      <div
        class="col-8"
      >
        <div
          class="card shadow-sm mt-3 border-0 "
        >
          <div
            class="card-body p-2"
          >
            <div
              class="row no-gutters"
            >
              <span
                class="badge badge-pill badge-success position-absolute lesson-card__container_completed"
              >
                <svg
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  style="height: 15px; margin-top: -2px;"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
                  />
                  <polyline
                    points="22 4 12 14.01 9 11.01"
                  />
                </svg>
                <span
                  class="mr-1"
                >
                  COMPLETED
                </span>
              </span>
              <div
                class="mw-100 col-2 mr-4"
              >
                <img
                  alt="js-0-cover.svg"
                  src="/curriculumAssets/lessonCoversSvg/js-0-cover.svg"
                />
              </div>
              <div
                class="col-9"
              >
                <h4
                  class="lesson-card__title font-weight-bold mt-3"
                >
                  lesson 1
                </h4>
                <div>
                  <div
                    class="d-inline-block mr-4"
                  >
                    <img
                      alt="icon-challenge"
                      class="mr-1"
                      src="/curriculumAssets/icons/icon-challenge.svg"
                    />
                    <span
                      class="lesson-card__icon-text"
                    >
                      1
                       CHALLENGES
                    </span>
                  </div>
                </div>
                <p
                  class="lesson-card__description mt-2"
                >
                  1st lesson
                </p>
              </div>
              <span
                class="position-absolute lesson-card__container_review"
              >
                <a
                  class="btn btn-sm bg-primary text-white"
                  href="https://c0d3.com/teacher/1"
                >
                  Review Submissions
                </a>
              </span>
            </div>
          </div>
        </div>
        <div
          class="card shadow-sm mt-3 border-0 "
        >
          <div
            class="card-body p-2"
          >
            <div
              class="row no-gutters"
            >
              <span
                class="badge badge-pill badge-success position-absolute lesson-card__container_completed"
              >
                <svg
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  style="height: 15px; margin-top: -2px;"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
                  />
                  <polyline
                    points="22 4 12 14.01 9 11.01"
                  />
                </svg>
                <span
                  class="mr-1"
                >
                  COMPLETED
                </span>
              </span>
              <div
                class="mw-100 col-2 mr-4"
              >
                <img
                  alt="js-1-cover.svg"
                  src="/curriculumAssets/lessonCoversSvg/js-1-cover.svg"
                />
              </div>
              <div
                class="col-9"
              >
                <h4
                  class="lesson-card__title font-weight-bold mt-3"
                >
                  lesson 4
                </h4>
                <div>
                  <div
                    class="d-inline-block mr-4"
                  >
                    <img
                      alt="icon-challenge"
                      class="mr-1"
                      src="/curriculumAssets/icons/icon-challenge.svg"
                    />
                    <span
                      class="lesson-card__icon-text"
                    >
                      1
                       CHALLENGES
                    </span>
                  </div>
                </div>
                <p
                  class="lesson-card__description mt-2"
                >
                  4th lesson
                </p>
              </div>
              <span
                class="position-absolute lesson-card__container_review"
              >
                <a
                  class="btn btn-sm bg-primary text-white"
                  href="https://c0d3.com/teacher/4"
                >
                  Review Submissions
                </a>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-4"
      >
        <div
          class="progress-card__container d-flex card shadow-sm mt-3 bg-primary text-white p-2 border-0"
        >
          <div
            class="card-body"
          >
            <div
              class="progress-card__progress-bar d-flex justify-content-center align-items-center rounded-circle"
            >
              <span
                class="font-weight-bold"
              >
                0
                %
              </span>
            </div>
            <h4
              class="progress-card__title mt-3"
            >
              You're off to a great start!
            </h4>
            <div
              class="mt-3"
            >
              <p>
                Start by setting up your development environment and then follow the lessons in the course.
              </p>
            </div>
            <a
              class="btn btn-light progress-card__button mt-2 text-primary"
              href="https://www.notion.so/Setup-Instructions-fc8f8fcc1376482ead839fa6b1034cb4#10ab6286f4d942c58bd330f68cb8cb95"
            >
              Setup Development Environment
            </a>
          </div>
        </div>
        <div
          class="announcement-card__container card shadow-sm mt-3 p-2 d-flex border-0"
        >
          <div
            class="card-body"
          >
            <h1
              class="announcement-card__title"
            >
              General Announcements
            </h1>
            <div
              class="mt-3"
            >
              <p
                class="font-weight-bold"
              >
                Updates and Guidelines
              </p>
              <p
                class="mt-4"
              >
                To make space for other students on our servers, your account will be deleted after 30 days of inactivity.
              </p>
              <p
                class="mt-4"
              >
                Take each lesson challenge seriously and do them over and over again until you can solve them. With the exception End to End, all challenges are questions and exercises taken from real interviews.
              </p>
              <p
                class="mt-4"
              >
                This lesson will not only prepare you for interviews, but it will also help teach you the skills that you need to become an effective engineer.
              </p>
              <p
                class="mt-4"
              >
                After completing Foundations of JavaScript, Variables & Functions, Array, Objects, End to End, HTML/CSS/JavaScript, React/GraphQL/SocketIO, you will be technically ready to contribute to our codebase.
              </p>
            </div>
          </div>
        </div>
        <div
          class="additional-resources__container mt-4 mb-0"
        >
          <h1
            class="additional-resources__title text-muted"
          >
            ADDITIONAL RESOURCES
          </h1>
          <hr
            class="my-2"
          />
          <a
            class="mx-3 "
            href="https://www.notion.so/Setup-Instructions-fc8f8fcc1376482ead839fa6b1034cb4"
          >
            Setup Instructions
          </a>
          <hr
            class="my-2"
          />
          <a
            class="mx-3 "
            href="#"
          >
            Github Basics
          </a>
          <hr
            class="my-2"
          />
          <a
            class="mx-3 "
            href="https://stackoverflow.com/"
          >
            Stack Overflow
          </a>
          <hr
            class="my-2"
          />
          <a
            class="mx-3 "
            href="https://chat.c0d3.com/"
          >
            Help & Support
          </a>
          <hr
            class="my-2"
          />
        </div>
      </div>
    </div>
  </div>
  <footer
    class="mt-5 font-weight-light text-center pb-5 text-muted"
  >
    Â© Copyright 2020 GarageScript
    <br />
    All Rights Reserved
  </footer>
</div>
`;

exports[`Curriculum Page Should render ellpsis when no data is present 1`] = `
<div>
  <h1>
    ...
  </h1>
</div>
`;

exports[`Curriculum Page Should render loading when loading 1`] = `
<div>
  <h1>
    Loading
  </h1>
</div>
`;
